### PR TITLE
feat: useClipboard hook

### PR DESCRIPTION
**Pull request by dougfabris**

### CODE REVIEW

File: packages/fuselage-hooks/src/useClipboard.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  onCopySuccess?: (e?: Event) => void;
+  onCopyError?: (e?: Event) => void;
+};
+
+export type UseClipboardReturn = {
+  copy: (e?: Event) => Promise<void>;
+  hasCopied: boolean;
+};
+
+export const useClipboard = (
```
Comment (tassoevan) [inline packages/fuselage-hooks/src/useClipboard.ts:23]:
```suggestion
type UseClipboardParams = {
  clearTime?: number;
  onCopySuccess?: (e?: Event) => void;
  onCopyError?: (e?: Event) => void;
};

export type UseClipboardReturn = {
  copy: (e?: Event) => Promise<void>;
  hasCopied: boolean;
};

/**
 * Hook to copy the passed content to the clipboard.
 *
 * @returns an object with the copy function and the hasCopied state
 * @public
 */
export const useClipboard = (
```

Doc comments have the inconvenient of requiring to be placed above the declaration they refer, without blank lines between them.

File: packages/fuselage-hooks/README.md [hunk 2]
```diff
@@ -101,6 +103,20 @@ Hook to catch which responsive design' breakpoints are active.
 
 Returns **[Array](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array)&lt;[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)>** an array of the active breakpoint names
 
+### UseClipboardParams
+
+Hook to copy the passed content to the clipboard.
```
Comment (tassoevan) [inline packages/fuselage-hooks/README.md:108]: This is wrong due to a misplaced doc comment.

**Review by tassoevan — CHANGES_REQUESTED** [review]

File: packages/fuselage-hooks/README.md (hunk 2 above)
Comment (dougfabris) [reply packages/fuselage-hooks/README.md]: New docs generated

File: packages/fuselage-hooks/src/useClipboard.ts (hunk 1 above)
Comment (dougfabris) [reply packages/fuselage-hooks/src/useClipboard.ts]: Nice to know about. I already fixed it

**Review by tassoevan — APPROVED** [review]

**Pull request merged**